python/taichi/main.py

```diff
@@ -104,11 +118,11 @@ def main(debug=False):
   elif mode == "test_cpp":
     return test_cpp()
   elif mode == "test":
-    if test_python() != 0:
+    if test_python(test_files=sys.argv[2:]) != 0 or len(sys.argv) > 2:
```

Comment:
I only did this for `test_python` because `test_cpp()` uses `sys.argv[2:]`:
https://github.com/taichi-dev/taichi/blob/62aaaca8216206022f1b93ecf8bfb4dfcd5b352b/python/taichi/main.py#L39

I like this change because we run python tests much more often. But then let's remove the `sys.args[2:]` usage in `test_cpp()`?  

Also, if we want to terminate earlier when the input is `ti test xx yy` (i.e. the `len(sys.argv) > 2` check), it should probably be put into a different branch, so that the program returns 0 instead of -1

Reply:
Maybe we can add test_files argument to test_cpp

Reply:
Agree, but i think we should switch to `argparse` before doing that, to avoid implement these arg parsing logics on our own...

Reply:
Using `argparse` is a good idea. Sorry about the old & messy `ti` implementation with lots of `if`s! :-)